src/atomicassets.cpp

```diff
@@ -327,6 +327,102 @@ ACTION atomicassets::forbidnotify(
     });
 }
 
+/**
+* Creates a swap offer for a collection
+* Acceptance functionality depends on the authorization being 'owner' or 'active' (7 days gate)
+*/
+
+ACTION atomicassets::createauswap(
+    name collection_name,
+    name new_author,
+    name permission
+) {
+    auto collection_itr = collections.require_find(collection_name.value,
+        "No collection with this name exists");
+
+    if (permission == name("owner")){
+        require_auth(permission_level{collection_itr->author, name("owner")});
+    } else {
+        check(permission == name("active"), 
+            "Only 'owner' or 'active' permissions can create an author swap");
+        require_auth(permission_level{collection_itr->author, name("active")});
+    }
+
+    check(authorswaps.find(collection_name.value) == authorswaps.end(), 
+        "Can't swap author's while an authorswap is underway for this collection");
+    
+    authorswaps.emplace(collection_itr->author, [&](auto &_author_swaps) {
+        _author_swaps.collection_name = collection_name;
+        _author_swaps.current_author = collection_itr->author;
+        _author_swaps.new_author = new_author;
+        _author_swaps.permission = permission;
+        _author_swaps.swap_date = eosio::current_time_point().sec_since_epoch();
+    });
+}
+
+/**
+* Accepts an author swap, with time constraints based on 'owner' or 'active' permissions used when creating the author swap
+* With default parameters, author swaps created by 'active' permissions can only be accepted after 1 week has passed
+* With default parameters, author swaps remain valid for up to 3 weeks
+*/
+
+ACTION atomicassets::acceptauswap(
+    name collection_name
+) {
+    auto collection_itr = collections.require_find(collection_name.value,
+        "No collection with this name exists");
+
+    auto author_swaps_itr = authorswaps.require_find(collection_name.value,
+        "No author swaps for this collection found");
+
+    // Just in case**
+    check(collection_itr->author == author_swaps_itr->current_author, 
+        "Current author mismatch");
+
+    uint32_t now = eosio::current_time_point().sec_since_epoch();
+
+    if (author_swaps_itr->permission == name("owner")){
+        require_auth(permission_level{author_swaps_itr->new_author, name("owner")});
```

Comment:
It doesn't make sense to me that the new author also has to be the owner permission